app/controllers/api/accounts_controller.rb

```diff
@@ -6,23 +6,44 @@
 #  https://github.com/puzzle/cryptopus.
 
 class Api::AccountsController < ApiController
+  helper_method :team
 
+  # GET /api/api_users
   def index
     authorize Account
     accounts = current_user.accounts
     accounts = find_accounts(accounts)
     render_json accounts
   end
 
+  # GET /api/api_users/:id
   def show
-    account = Account.find(params[:id])
     authorize account
     account.decrypt(decrypted_team_password(account.group.team))
     render_json account
   end
 
+  # PATCH /api/api_users/:id?Query
+  def update
+    update_account
```

Comment:
so you are updating the account before authorizing it ? 
should be authorization always be the first thing to do ?
what does authorize do exactly ?

Reply:
The authorize Method is a function provided by the pundit gem -  when the method get's called, pundit looks in the same named policy class account. This, in this case is the account_policy class. 
Pundit therefore look if the user is authorized, if not it wil throw an execption. You should also do other account related authorization things, such as the permitted attributes method which checks the allowed attributes in an Object. In my opinion the authorization should always be done before the model is getting updated, but only if an authorization is necessary.